cli/azd/pkg/project/dotnet_importer.go

```diff
@@ -101,7 +102,12 @@ func (ai *DotNetImporter) ProjectInfrastructure(ctx context.Context, svcConfig *
 		return nil, fmt.Errorf("generating app host manifest: %w", err)
 	}
 
-	files, err := apphost.BicepTemplate(manifest)
+	useResourceGroupScopedDeployment := false
+	if v, err := strconv.ParseBool(os.Getenv("AZD_DEBUG_DOTNET_APPHOST_USE_RESOURCE_GROUP_DEPLOYMENTS")); err == nil {
```

Comment:
This isn't the right way to understand the user intention here.  In cases where IaC for a project already exists, we use the bicep to determine if the user wanted a resource group scoped deployment or a subscription scoped deployment.  In Aspire, since we /generate/ the infrastructure - we have to ask the user which version they'd like.  Right now that's done with this debug env var - but in an ideal world I think it gets captured in `azure.yaml` (as a service specific `infra` property, like `deploymentScope`?) or it is purely a concern of the code in the AppHost and it will generate the right bicep for us.

Need to think a bit more about

Reply:
@vhvb1989 - Interested in your thoughts about how we might want to have the user express their intention here.  I think you have a clearer sense than I about where things are headed as we move more of the IaC generation out of `azd` and into the Aspire AppHost.

Reply:
I had actually started with resuing `AZURE_RESOURCE_GROUP` but realized that would force the user to set the value and then would prevent `azd provision` for asking for it - which means in cases where the rg did not exist and you wanted `azd` to create it via prompting, we'd now fail.

Reply:
> that would force the user to set the value and then would prevent azd provision for asking for it 

IIRC, If the value is an empty string, azd would prompt. 
User can set the value when they know what rg to use, or use an empty string to let azd o prompt for it.